GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerTestCase/PersistedModifiedPathsTests.cs

```diff
@@ -116,8 +155,39 @@ public void ExcludeSparseFileSavedAfterRemount(FileSystemRunner fileSystem)
             modifiedPathsDatabase.ShouldBeAFile(fileSystem);
             using (StreamReader reader = new StreamReader(File.Open(modifiedPathsDatabase, FileMode.Open, FileAccess.Read, FileShare.ReadWrite)))
             {
-                reader.ReadToEnd().ShouldEqual(ExpectedModifiedFilesContents);
+                reader.ReadToEnd().Split(new char[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries).OrderBy(x => x)
+                    .ShouldMatchInOrder(ExpectedModifiedFilesContents.OrderBy(x => x));
+            }
+        }
+
+        private string CreateDirectory(FileSystemRunner fileSystem, string relativePath)
+        {
+            string tempFolder = this.Enlistment.GetVirtualPathTo(relativePath);
+            fileSystem.CreateDirectory(tempFolder);
+            tempFolder.ShouldBeADirectory(fileSystem);
+            return tempFolder;
+        }
+
+        private string CreateFile(FileSystemRunner fileSystem, string relativePath)
+        {
+            string tempFile = this.Enlistment.GetVirtualPathTo(relativePath);
+            fileSystem.WriteAllText(tempFile, $"Contents for the {relativePath} file");
+            tempFile.ShouldBeAFile(fileSystem);
+            return tempFile;
+        }
+
+        private void ValidateModifiedPathsDoNotContain(FileSystemRunner fileSystem, params string[] paths)

```

Comment:
I think I can use it but a little bit differently than intended because it is just searching the entire string from the modified paths so if I have a line like
Temp/dir1/file.txt
Temp/, dir1/, Temp/dir1/ would all be found even though the modified paths does not contain those paths directly which is what I am looking for.
Looking to use it like this which should match or not match a line in the modified paths.
```
GVFSHelpers.ModifiedPathsShouldNotContain(fileSystem, this.Enlistment.DotGVFSRoot, paths.Select(x => $"A {x}" + Environment.NewLine).ToArray());
GVFSHelpers.ModifiedPathsShouldNotContain(fileSystem, this.Enlistment.DotGVFSRoot, paths.Select(x => $"D {x}" + Environment.NewLine).ToArray());
```